packages/flutter/lib/src/widgets/expansible.dart

```diff
@@ -0,0 +1,367 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport 'package:flutter/material.dart';
+library;
+
+import 'basic.dart';
+import 'framework.dart';
+import 'page_storage.dart';
+import 'ticker_provider.dart';
+import 'transitions.dart';
+
+/// The type of the callback that returns the header or body of an [Expansible].
+///
+/// The `animation` property exposes the underlying expanding or collapsing
+/// animation, which has a value of 0 when the [Expansible] is completely
+/// collapsed and 1 when it is completely expanded. This can be used to drive
+/// animations that sync up with the expanding or collapsing animation, such as
+/// rotating an icon.
+///
+/// See also:
+///
+///   * [Expansible.headerBuilder], which is of this type.
+///   * [Expansible.bodyBuilder], which is also of this type.
+typedef ExpansibleComponentBuilder =
+    Widget Function(BuildContext context, Animation<double> animation);
```

Comment:
I'm concerned about the animation here because it isn't compatible with simulations, which is heavily used by iOS components (spring). However I don't have a good suggestion...